app/forms/personal_details_form.rb

```diff
@@ -81,6 +81,12 @@ def nationality_ids
       end
   end
 
+  def missing_fields
+    (fields.select { |k, v| k if v.blank? }.keys & FIELDS).tap do |field|
+      field << :date_of_birth unless date_of_birth.is_a?(Date)
+    end
+  end
```

Comment:
```suggestion
  def missing_fields
    if !self.valid?(:missing_data)
      self.errors.attribute_names      
    end  
  end
```

This just looks like another form of validation to me.

Reply:
Good shout have updated